src/services/userTransform/userTransform.service.ts

```diff
@@ -0,0 +1,172 @@
+import groupBy from "lodash/groupBy";
+import isEmpty from "lodash/isEmpty";
+import { userTransformHandler } from "../../routerUtils";
+import {
+  Library,
+  ProcessorRequest,
+  ProcessorResponse,
+  UserTransformResponse,
+  UserTransfromServiceResponse
+} from "../../types/index";
+import { RespStatusError, RetryRequestError } from "../../util/utils";
+import { getMetadata, isNonFuncObject } from "../../v0/util";
+import logger from "../../logger";
+
+export default class UserTransformService {
+  public static async userTransformRoutine(
+    events: ProcessorRequest[]
+  ): Promise<UserTransfromServiceResponse> {
+    let retryStatus = 200;
+    const groupedEvents: Object = groupBy(
+      events,
+      (event: ProcessorRequest) =>
+        `${event.metadata.destinationId}_${event.metadata.sourceId}`
+    );
+    const transformedEvents = [];
+    let librariesVersionIDs = [];
+    if (events[0].libraries) {
+      librariesVersionIDs = events[0].libraries.map(
+        (library: Library) => library.VersionID
+      );
+    }
+    const responses = await Promise.all<any>(
+      Object.entries(groupedEvents).map(async ([dest, destEvents]) => {
+        logger.debug(`dest: ${dest}`);
+        const eventsToProcess = destEvents as ProcessorRequest[];
+        const transformationVersionId =
+          eventsToProcess[0]?.destination?.Transformations[0]?.VersionID;
+        const messageIds = eventsToProcess.map(ev => {
+          return ev.metadata?.messageId;
+        });
+
+        const commonMetadata = {
+          sourceId: eventsToProcess[0]?.metadata?.sourceId,
+          destinationId: eventsToProcess[0]?.metadata.destinationId,
+          destinationType: eventsToProcess[0]?.metadata.destinationType,
+          messageIds
+        };
+
+        const metaTags =
+          eventsToProcess.length && eventsToProcess[0].metadata
+            ? getMetadata(eventsToProcess[0].metadata)
+            : {};
+
+        if (transformationVersionId) {
```

Comment:
handle not first using `if (!transformationVersionId) {` then you can avoid nesting for other case, more readable.

Reply:
fixed